framework/src/base/MooseApp.C

```diff
@@ -3503,24 +3527,29 @@ MooseApp::determineLibtorchDeviceType(const MooseEnum & device_enum) const
   {
 #ifdef __linux__
     if (!torch::cuda::is_available())
-      mooseError("--libtorch-device=cuda: CUDA is not available");
+      mooseError(
+          "--compute-device=cuda: CUDA support is not available in the linked libtorch library");
```

Comment:
If the user passed `--libtorch-device`, this error might be a little confusing although they'd probably figure it out in combination with the deprecation message. Regardless I might make the beginning of these strings dependent on which param was set by the user

Reply:
Ok, I made that change in a separate easily revertible commit for when we remove `--libtorch-device`.